ckanext/aircan_connector/util.py

```diff
@@ -0,0 +1,94 @@
+# encoding: utf-8
+import os
+import requests
+from ckan.common import config
+import logging
+import tempfile
+import math
+import hashlib
+import messytables
+
+log = logging.getLogger(__name__)
+
+SSL_VERIFY = True
+if not SSL_VERIFY:
+    requests.packages.urllib3.disable_warnings()
+
+CHUNK_SIZE = 16 * 1024  # 16kb
+DOWNLOAD_TIMEOUT = 30
+
+def get_file_headers(resource, api_key):
+
+    url = resource.get('url')
+
+    # fetch the resource data
+    log.info('Fetching from: {0}'.format(url))
+    tmp_file = get_tmp_file(url)
+    length = 0
+
+    try:
+        headers = {}
+        if resource.get('url_type') == 'upload':
+            # If this is an uploaded file to CKAN, authenticate the request,
+            # otherwise we won't get file from private resources
+            headers['Authorization'] = api_key
+        log.info("headers: {}".format(headers))
+        log.info("SSL_VERIFY: {}".format(SSL_VERIFY))
+        response = requests.get(
+            url,
+            headers=headers,
+            timeout=DOWNLOAD_TIMEOUT,
+            verify=SSL_VERIFY,
+            stream=True,  # just gets the headers for now
+        )
+        response.raise_for_status()
+
+        # download the file to a tempfile on disk
+        for chunk in response.iter_content(CHUNK_SIZE):
+            length += len(chunk)
+            tmp_file.write(chunk)
+
+        log.info('Downloaded ok - %s', printable_file_size(length))
```

Comment:
Quick question: Why do you need the file_size here? Is it just a check to see if the file is downloaded? If you are just using it in the logs it may not be necessary

Reply:
so it's not relevant for us at all anymore, correct? May I close this PR?

Reply:
@mbeilin We'll have to get the headers (no type inference) if they exist. We cannot have something hard-coded like `"schema_fields_array": [ "FID", "Mkt-RF", "SMB", "HML", "RF" ],` .  I'm just a bit lost in the implementation. Like, we have a CSV file already; can't we just get its headers (if any)?

Reply:
@hannelita I think it should be provided by the user (tableschema.json with `[ "FID", "Mkt-RF", "SMB", "HML", "RF" ]` ) together with the resource file(csv). otherwise, we are not triggering the DAG

Reply:
All right, so we can close the PR